### ISSUE TITLE
FR Rule: Prevent snapshot inside wait

### ISSUE THREAD

**Issue by SBoudrias** [testing-library/eslint-plugin-testing-library#214]
I wonder if there'd be a way to make sure there's no snapshot done inside wait conditions.

Example invalid usage:

```js
await wait(() => {
  expect(foo).toMatchSnapshot();
});
```

Valid:
```js
expect(foo).toMatchSnapshot();
```

This is between jest and and testing-library, but the depreciated `wait()` functions allow tests like this to be written and lead to flakey cases.

**Comment by Belco90** [comment]
Hey Simon, sorry for the late response!

This sounds like an interesting rule, but I don't see the problem 100% clear. Could you elaborate more around the issues this can cause please? Thanks.

**Comment by SBoudrias** [comment]
Yes of course!

The problem is that `wait()` runs in a timer loop. So it'll retry every n amount of time.

If there's a snapshot inside the wait condition, jest will generate one snapshot per loop.

The problem then is the amount of loop ran until the condition is met will vary between different computers (or CI machines.) This leads to tests that will regenerate a lot of snapshots until the condition is match when devs run those tests locally updating the snapshots; e.g devs cannot run `jest -u` locally or it'll generate a lot of invalid snapshots who'll fail during CI.

This would be fine if snapshots were generated manually; but auto-update causes conflicts.

**Comment by Belco90** [comment]
I see. Yeah, didn't think about the auto update. This can definitely be a nice rule, something like `no-wait-snapshot`. Are you interested into give it a try?
